waspc/src/Wasp/Generator/Setup.hs

```diff
@@ -9,13 +9,21 @@ import Wasp.Generator.Common (ProjectRootDir)
 import qualified Wasp.Generator.DbGenerator as DbGenerator
 import Wasp.Generator.Monad (GeneratorError (..), GeneratorWarning (..))
 import Wasp.Generator.NpmInstall (ensureNpmInstall)
+import Wasp.Message (MessageType (..), SendMessage)
 
-runSetup :: AppSpec -> Path' Abs (Dir ProjectRootDir) -> IO ([GeneratorWarning], [GeneratorError])
-runSetup spec dstDir = do
-  -- TODO: Use a monad transformer here in the future for better shortcircuiting behavior.
+runSetup :: AppSpec -> Path' Abs (Dir ProjectRootDir) -> SendMessage -> IO ([GeneratorWarning], [GeneratorError])
+runSetup spec dstDir sendMessage = do
+  sendMessage $ Start "Starting npm install..."
   (npmInstallWarnings, npmInstallErrors) <- ensureNpmInstall spec dstDir
   if null npmInstallErrors
     then do
+      sendMessage $ Success "Successfully completed npm install."
+      sendMessage $ Start "Setting up database..."
       (dbGeneratorWarnings, dbGeneratorErrors) <- DbGenerator.postWriteDbGeneratorActions spec dstDir
+      if null dbGeneratorErrors
+        then sendMessage $ Success "Database successfully set up..."
+        else sendMessage $ Error "Could not set up database."
       return (npmInstallWarnings ++ dbGeneratorWarnings, dbGeneratorErrors)
-    else return (npmInstallWarnings, npmInstallErrors)
+    else do
+      sendMessage $ Error "Failed npm install!"
```

Comment:
The reason I wrote it this way was because I wanted to start the sentence with a capital, instead of starting it with lowercase because the npm command is lower cased. But it's not a big deal to me either way so I'll take this change.